lib/saluki-components/src/sources/dogstatsd/mod.rs

```diff
@@ -298,6 +321,10 @@ impl SourceBuilder for DogStatsDConfiguration {
             }),
             codec,
             context_resolver,
+            pre_filters: vec![Arc::new(EnablePayloadsFilter::new(
```

Comment:
Flip this around, so it's `Arc<Vec<dyn Filter + Send + Sync>>`, so that we're only cloning the the `Arc<...>` when we create a new listener/handler/stream, instead of actually cloning the `Vec<...>`.